### PR TITLE
[T5A1][W13-B1]Yong Zhi Yuan

### PR DESCRIPTION
**Pull request by Zhiyuan-Amos**

### CODE REVIEW

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -73,13 +76,48 @@ Use case ends.
...
   Use case resumes at step 2
+  
+#### Use case: Edit existing tag
+
+**MSS**
+
+1. User requests to list persons
+2. AddressBook shows a list of persons
+3. User requests to edit details of a specific person in the list
+4. User edits existing tag of the person
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: there is something missing, right? user requests to edit details, so address book should ask which detail first, then user edits the tag.
use case should indicate the complete system-user interaction explicitly.

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -73,13 +76,48 @@ Use case ends.
 
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
+  
+#### Use case: Edit existing tag
+
+**MSS**
+
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: the activity was updated to rename the tag. please take a look at the description now.

File: doc/DeveloperGuide.md [hunk 3]
```diff
@@ -73,13 +76,48 @@ Use case ends.
...
+> 3a1. AddressBook shows an error message <br>
+  Use case resumes at step 2
+  
+4a. The existing tag is invalid
+
+> 4a1. AddressBook shows an error message, requiring user to enter a valid tag <br>
+  Use case resumes at step 4
+
+4b. The user has no existing tags
+
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: may be, if user doesn't have a tag, you can updated the person with the given tag, no?

File: doc/DeveloperGuide.md [hunk 4]
```diff
@@ -73,13 +76,48 @@ Use case ends.
...
+> 4b1. AddressBook shows an error message, requiring user to select another index <br>
+  Use case resumes at step 2
 
 ## Appendix C : Non Functional Requirements
 
 1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java 8 or higher installed.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Response time per command should be less than 0.5s.
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: :+1:  very precise NFR; only one?

**Comment by okkhoy** [comment]
@Zhiyuan-Amos 
Some comments added. Please ack & close the PR after reading comments.

File: doc/DeveloperGuide.md (hunk 3 above)
Comment (Zhiyuan-Amos) [inline doc/DeveloperGuide.md]: Hello @okkhoy , can I know what is the proper way to indent the different possible choices? I just added an additional space as the indentation.

> 4b1. AddressBook shows an error message, asking user if user wants to add a tag instead <br>
>  4b1a. User selects yes
>   4b1a1. AddressBook prompts user to enter the new tag
>   4b1a2. User enters the new tag
>   4b1a3. AddressBook saves the tag of the person
    Use case ends
>  4b1b. User selects no
   Use case ends

Thanks Akshay! :)

**Pull request closed without merging**